src/Chapter4.hs

```diff
@@ -653,10 +652,13 @@ instance Monad List where
 -- This way works with the monad operators, but I prefer using Functor and Applicative
 -- for this particular solution (below)
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM m1 m2 = m1 >>= (\a -> m2 >>= (\b -> pure $ a && b))
-
-andM' :: (Monad m) => m Bool -> m Bool -> m Bool
-andM' m1 m2 = (&&) <$> m1 <*> m2
+andM m1 m2 =
+  m1
+    >>= ( \a ->
+            if a == False
+              then pure False
+              else m2 >>= (\b -> pure $ a && b)
```

Comment:
As `a` is bool, you can just have condition on that:

```haskell
if a 
then m2 >>= ...
else pure False
```

Reply:
Thanks, I don't know why my brain wasn't working properly :P